ast/src/main/java/com/graphicsfuzz/common/ast/type/QualifiedType.java

```diff
@@ -79,6 +80,34 @@ public void replaceQualifier(TypeQualifier oldQualifier, TypeQualifier newQualif
     qualifiers.add(newQualifier);
   }
 
+  /**
+   * Sets or replaces location qualifier.
+   *
+   * @param location Location to set in the location qualifier.
+   */
+  public void setLocationQualifier(int location) {
+    Optional existingLayoutQualifierSequence = Optional.empty();
+    List<LayoutQualifier> qualifierList = new ArrayList<LayoutQualifier>();
+    qualifierList.add(new LocationLayoutQualifier(location));
+    for (TypeQualifier t : qualifiers) {
+      if (t instanceof LayoutQualifierSequence) {
+        if (existingLayoutQualifierSequence.isPresent()) {
+          throw new RuntimeException("More than one layout qualifier sequence found");
+        }
+        existingLayoutQualifierSequence = Optional.of(t);
+        for (LayoutQualifier l : ((LayoutQualifierSequence) t).getLayoutQualifiers()) {
```

Comment:
I would change this to:

        for (LayoutQualifier l : existingLayoutQualifierSequence.get().getLayoutQualifiers()) 

to avoid another case (since you will now need one for t).

Reply:
Done